hudi-utilities/src/main/java/org/apache/hudi/utilities/HoodieCompactor.java

```diff
@@ -269,6 +269,7 @@ private int doCompact(JavaSparkContext jsc) throws Exception {
         }
       }
       HoodieWriteMetadata<JavaRDD<WriteStatus>> compactionMetadata = client.compact(cfg.compactionInstantTime);
+      cleanAfterCompact(client);
       return UtilHelpers.handleErrors(compactionMetadata.getCommitMetadata().get(), cfg.compactionInstantTime);
```

Comment:
Can we just execite a synchoronous cleaning for the offline compaction and clustering, it does not make sense to make the cleaning async because the whole job in executed as a batch.